compiler/rustc_trait_selection/src/error_reporting/traits/suggestions.rs

```diff
@@ -5178,6 +5178,46 @@ impl<'a, 'tcx> TypeErrCtxt<'a, 'tcx> {
             _ => {}
         }
     }
+
+    pub fn suggest_remove_brackets_from_range(
+        &self,
+        main_trait_predicate: ty::PolyTraitPredicate<'tcx>,
+        leaf_trait_predicate: ty::PolyTraitPredicate<'tcx>,
+        span: Span,
+        err: &mut Diag<'_>,
+    ) {
+        let is_trait_range_bounds =
+            match self.tcx.get_diagnostic_name(leaf_trait_predicate.def_id()) {
+                Some(trait_name) if trait_name == sym::RangeBounds => true,
+                _ => false,
+            };
+        if !is_trait_range_bounds {
+            return;
+        }
+        if let ty::Array(inner_type, size_value) =
+            main_trait_predicate.self_ty().skip_binder().kind()
+        {
+            let is_inner_type_range = if let ty::Adt(adt_def, _) = inner_type.kind()
+                && self.tcx.is_range(adt_def.did())
+            {
+                true
+            } else {
+                false
+            };
```

Comment:
Nit: Can also be simplified:
```suggestion
            let ty::Adt(adt_def, _) = inner_type.kind() else {
                return;
            }
            if !self.tcx.is_range(adt_def.did()) {
                return;
            }
```

Similarly, you can use early return for the other checks too, I'd prefer that.

Alternatively, if you want to keep the `is_inner_type_range` variable, do:

```suggestion
            let is_inner_type_range = matches!(
                inner_type.kind(),
                ty::Adt(adt_def, _) if self.tcx.is_range(adt_def.did())
            );
```